one/src/lib.rs

```diff
@@ -371,7 +380,16 @@ impl Daemon {
             max_conns_pending_in: opts.max_conns_pending_in,
             max_conns_per_peer: opts.max_conns_per_peer,
             idle_connection_timeout: Duration::from_millis(opts.idle_conns_timeout_ms),
-            bootstrap_peers: opts.network.bootstrap_addresses(),
+            // Add extra bootstrap addresses to the list of official bootstrap addresses, so that our bootstrap nodes
+            // are always included.
+            bootstrap_peers: [
+                opts.network.bootstrap_addresses(),
+                opts.extra_bootstrap_addresses
```

Comment:
I believe there is an `extend` or maybe `append` method on a vector that can take in an iterator. Would simplify this code a bit.